docs/dev/topology-over-raft.md

```diff
@@ -468,8 +471,11 @@ There are also a few static columns for cluster-global properties:
 - `committed_cdc_generations` - the IDs of the committed CDC generations
 - `unpublished_cdc_generations` - the IDs of the committed yet unpublished CDC generations
 - `global_topology_request` - if set, contains one of the supported global topology requests
+- `global_topology_request_id` - if set, contains global topology request's id, which is a new group0's state id
```

Comment:
This new global_topology_request_id deserves some more descriptive comment which should specifically point to the difference between it and the existing global_topology_request. It can be a patch comment, or code comment

Reply:
`global_topology_request_id` stores any sort of ID we want to use to track the lifetime of `global_topology_request`, it can be a string, UUID, integer, anything. Since we use `topology_request_tracking_mutation_builder` to track it, and the convention is to use group0's state id, I wrote it's:
> a new group0's state id

And that's it. Now you must help me come up with a more descriptive comment :)

Reply:
Ah, so you need to keep a reference from `system.topology`'s record to `system.topology_requests`'s record, right?

Reply:
Correct.

Reply:
But what is this reference needed for? The way I see it:

- `cql3::statements::alter_keyspace_statement::prepare_schema_mutations` adds record to `system.topology` and `system.topology_requests` and the former references the latter, and the latter has `keyspace_rf_change` request set
- next, topology coordinator gets into `handle_global_request`, notices the `keyspace_rf_change` set, then updates the `system.topology_requests` record by marking it as "done". The marking needs to know the record primary key, which is the golobal_topology_request_id itself

That's it, no other usage of this reference.

So the question is -- can we update update topology request as "done" by using its ID obtained from the `_topo_sm._topology.global_request` inside handle_topology_request itself? What's the need in keeping this ID in system.topology table?

Reply:
@xemul This is an example entry from the table keeping track of ongoing topology requests (in this specific case it's actually a global topo `rf_change` req):
```
cqlsh> select * from system.topology_requests;

@ Row 1
-----------------+---------------------------------------------------------------------------------
 id              | 764646e0-1449-11ef-88fb-173b23510037
 done            | True
 end_time        | 2024-05-17 12:32:03.053000+0000
 error           | null
 initiating_host | e4f7fbf0-e36e-41d5-8bb7-cc0f06ea88e6
 start_time      | 2024-05-17 12:32:02.715000+0000
```

In order to execute ALTER, query_processor and topology_coordinator need to both track the same req and the only "handler" to this request is the `id`. query_processor has this id, since it initiates the req, but topology_coordinator has to get this id from somewhere, and this "somewhere" is `system.topology` table.

> So the question is -- can we update update topology request as "done" by using its ID obtained from the `_topo_sm._topology.global_request` inside handle_topology_request itself? What's the need in keeping this ID in system.topology table?

`system.topology.global_topology_request` is either set to `null` or a string saying which global topo is being executed; this string can be e.g. `"keyspace_rf_change"`. It's not enough to use this string as an id in `system.topology_requests`, because `system.topology_requests` can track/store more than 1 ALTER statement, which makes this string `"keyspace_rf_change"` not unique. We have to have a unique identifier for each ALTER (and for any other req, for that matter), and we cannot reuse any data from `system.topology`, unfortunately.

Reply:
Le me know if that's a sufficient motivation or do you want to approach this issue from another angle.

Reply:
> topology_request_tracking_mutation_builder to track it, and the convention is to use group0's state id

It has to be UUID since this is what table's schema has, so why making the ID to be a string? I also do not understand why waiting is done outside of the "alter" code (this is apparently the reason why passing the id is needed), but may be I will see it in future patches.

Reply:
Ah, I see. The `handle_global_request()` handles a entry from `system.topology` and needs to update one from `system.topolgy_requests`